### ISSUE TITLE
engine: continue rewrite of tor experiment

### ISSUE THREAD

**Issue by bassosimone** [ooni/probe#1688]
Depends on: https://github.com/ooni/probe/issues/1687

Continues: https://github.com/ooni/probe/issues/1596

Requires: significant issues cleanup and code-base cleanup once done (maybe as part of another issue?)

**Comment by bassosimone** [comment]
This is now done!
